subscribe_test.go

```diff
@@ -259,6 +263,51 @@ func TestSubscribe(t *testing.T) {
 	testSubscribe(t, 3)
 }
 
+func testSubscribeLogs(t *testing.T, hub *Hub, payload interface{}) {
+	t.Helper()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	req := httptest.NewRequest(http.MethodGet, defaultHubURL+"?topic=http://example.com/reviews/{id}", nil).WithContext(ctx)
+	req.AddCookie(&http.Cookie{Name: "mercureAuthorization", Value: createDummyAuthorizedJWTWithPayload(hub, roleSubscriber, []string{"http://example.com/reviews/22"}, payload)})
+
+	w := &responseTester{
+		expectedStatusCode: http.StatusOK,
+		expectedBody:       ":\n",
+		t:                  t,
+		cancel:             cancel,
+	}
+
+	hub.SubscribeHandler(w, req)
+}
+
+func TestSubscribeWithDebug(t *testing.T) {
+	core, logs := observer.New(zapcore.DebugLevel)
+	var payload interface{}
+	json.Unmarshal([]byte("{\"bar\": \"baz\", \"foo\": \"bar\"}"), payload)
+
+	testSubscribeLogs(t, createDummy(
+		WithDebug(),
+		WithLogger(zap.New(core)),
+	), payload)
+
+	assert.True(t, logs.FilterMessage("New subscriber").FilterField(zap.Reflect("payload", payload)).Len() == 1)
+}
+
+func TestSubscribeWithoutDebug(t *testing.T) {
+	core, logs := observer.New(zapcore.DebugLevel)
+
+	var payload struct {
+		Bar string `json:"bar"`
+	}
+	payload.Bar = "baz"
```

Comment:
You can use inline initialization here:

```suggestion
	var payload struct {
		Bar string `json:"bar"`
	}{Bar: "baz"}
```